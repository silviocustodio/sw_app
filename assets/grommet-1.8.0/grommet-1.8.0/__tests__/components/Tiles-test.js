// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Tiles from '../../src/js/components/Tiles';
import Tile from '../../src/js/components/Tile';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Tiles', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Tiles>
        <Tile>
          First
        </Tile>
        <Tile>
          Second
        </Tile>
        <Tile>
          Third
        </Tile>
      </Tiles>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders selected tiles', () => {
    const component = renderer.create(
      <Tiles selected={[0, 1]}>
        <Tile>
          First
        </Tile>
        <Tile>
          Second
        </Tile>
        <Tile>
          Third
        </Tile>
      </Tiles>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
