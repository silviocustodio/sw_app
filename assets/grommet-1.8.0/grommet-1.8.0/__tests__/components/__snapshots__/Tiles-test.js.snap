// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tiles has correct default options 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--responsive grommetux-box--pad-small grommetux-box--wrap grommetux-tiles grommetux-tiles--flush"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    First
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    Second
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    Third
  </div>
</div>
`;

exports[`Tiles renders selected tiles 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--responsive grommetux-box--pad-small grommetux-box--wrap grommetux-tiles grommetux-tiles--flush"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-tile--selected grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    First
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-tile--selected grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    Second
  </div>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none grommetux-tile grommetux-none-hover-color-index-disabled grommetux-tile--hover-border-small"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    Third
  </div>
</div>
`;
